lib/init.js

```diff
@@ -15,17 +17,32 @@ init.usage = usage(
   'init',
   '\nnpm init [--force|-f|--yes|-y|--scope]' +
   '\nnpm init <@scope> (same as `npx <@scope>/create`)' +
-  '\nnpm init [<@scope>/]<pkg> (same as `npx [<@scope>/]create-<pkg>`)'
+  '\nnpm init [<@scope>/]<name> (same as `npx [<@scope>/]create-<name>`)'
 )
 
 function init (args, cb) {
   if (args.length) {
     var NPM_PATH = path.resolve(__dirname, '../bin/npm-cli.js')
     var initerName = args[0]
-    var packageName = /^@[^/]+$/.test(initerName)
-      ? initerName + '/create'
-      : initerName.replace(/^(@[^/]+\/)?/, '$1create-')
-
+    var packageName = initerName
+    if (/^@[^/]+$/.test(initerName)) {
+      packageName = initerName + '/create'
+    } else {
+      var req = npa(initerName)
+      if (isRegistry(req)) {
+        packageName = req.name.replace(/^(@[^/]+\/)?/, '$1create-')
+        if (req.rawSpec) {
+          packageName += '@' + req.rawSpec
+        }
+      } else {
+        log.warn(
```

Comment:
This should probably be an actual error. Maybe with `err.code === 'EUNSUPPORTED'` or something? 🤔

Reply:
Ah ok!